src/sentry/static/sentry/app/components/search/sources/routeSource.tsx

```diff
@@ -0,0 +1,181 @@
+import replaceRouterParams from 'app/utils/replaceRouterParams';
+import withLatestContext from 'app/utils/withLatestContext';
+import {NavigationItem} from 'app/views/settings/types';
+import {Organization, Project} from 'app/types';
+
+type Config =
+  | typeof accountSettingsNavigation
+  | typeof organizationSettingsNavigation
+  | typeof projectSettingsNavigation;
+
+// XXX(epurkhiser): We use the context in mapFunc to handle both producing the
+// NavigationSection list AND filtering out items in the sections that should
+// not be shown using the `show` attribute of the NavigationItem
+type Context = Parameters<Extract<Config, Function>>[0] &
+  Parameters<Extract<NavigationItem['show'], Function>>[0];
+
+/**
+ * navigation configuration can currently be either:
+ *
+ *  - an array of {name: string, items: Array<{NavItem}>} OR
+ *  - a function that returns the above
+ *    (some navigation items require additional context, e.g. a badge based on
+ *    a `project` property)
+ *
+ * We need to go through all navigation configurations and get a flattened list
+ * of all navigation item objects
+ */
+const mapFunc = (config: Config, context: Context | null = null) =>
+  (Array.isArray(config)
+    ? config
+    : context !== null
+    ? config(context)
+    : []
+  ).map(({items}) =>
+    items.filter(({show}) =>
+      typeof show === 'function' && context !== null ? show(context) : true
+    )
+  );
+
+type ItemProps = {
+  resultType: 'route';
+  sourceType: 'route';
+  to: string;
+};
+
+type Result = {
+  item: NavigationItem & ItemProps;
+  matches: FuseResultWithMatches<NavigationItem>['matches'];
+  score: number;
+};
+
+type RenderProps = {
+  isLoading: boolean;
+  /**
+   * Unused in this source
+   */
+  allResults: object[];
+  /**
+   * Matched results
+   */
+  results: Result[];
+};
+
+type DefaultProps = {
+  /**
+   * Fuse configuration for searching NavigationItem's
+   */
+  searchOptions: FuseOptions<NavigationItem>;
+};
+
+type Props = WithRouterProps &
+  DefaultProps & {
+    organization: Organization;
+    project: Project;
+    /**
+     * The string to search the navigation routes for
+     */
+    query: string;
+    /**
+     * Render function that renders the route matches
+     */
+    children: (props: RenderProps) => React.ReactNode;
+  };
+
+type State = {
+  /**
+   * A Fuse instance configured to search NavigationItem's
+   */
+  fuzzy: null | Fuse<NavigationItem, FuseOptions<NavigationItem>>;
+};
+
+class RouteSource extends React.Component<Props, State> {
+  static defaultProps: DefaultProps = {
+    searchOptions: {},
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    this.createSearch();
+  }
+
+  state: State = {
+    fuzzy: null,
+  };
+
+  componentDidUpdate(prevProps: Props) {
+    if (
+      prevProps.project === this.props.project &&
+      prevProps.organization === this.props.organization
```

Comment:
I believe this won't work as `project` and `organization` are of type object. Are you sure that `prevProps` and `this.props` point to the same object instance? Maybe you should use the lodash function "isEqual" here 😉

Reply:
idk this is what was already here haha. I think this is fine as long as the object instance is the same

Reply:
Yeah this should be the same object instance unless project/org changes